templates/include/prism/ast.h.erb

```diff
@@ -116,6 +116,21 @@ static const pm_node_flags_t PM_NODE_FLAG_COMMON_MASK = (1 << (PM_NODE_FLAG_BITS
  */
 #define PM_NODE_TYPE_P(node, type) (PM_NODE_TYPE(node) == (type))
 
+/**
+ * Sets the given flag on the given node.
+ */
+#define PM_NODE_FLAG_SET(node, flag) (((pm_node_t *)(node))->flags |= (flag))
```

Comment:
Some of the `node`s passed to these macros are specific node types and others were of type `pm_node_t *`. That meant we need to do `->base.flags` for the former, but only `->flags` for the latter, which broke the symmetry.

Since casting a specific node type should always give the `base` pointer, this cast should do the right thing and allow us to access flags using just `->flags`.